### PR TITLE
Handle duplicate attribute nodes

### LINKED ISSUE #9: Duplicate Entries in modules_to_attributes.json
There are two entries in `modules_to_attributes.json` which have the same value for the path key.  The duplicated value is: `"sr-document-content:0040a730:00700022"`.

We should investigate this, as it appears to be an error.

We should also check for other duplicates.

### PR DESCRIPTION
**Pull request by russellkan**
Done:
- Parse conditional from "include" statements in macro rows
- Add new processing module to merge attribute nodes with identical paths
- Add a few standard workarounds
- Add e2e test to check for duplicate paths
- Add unit tests for new processing module
- Close #9

### CODE REVIEW

File: dicom_standard/postprocess_merge_duplicate_nodes.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+def merge_duplicate_nodes(module_attr_list):
+    path_list = [d['path'] for d in module_attr_list]
+    duplicate_paths = [k for k, v in Counter(path_list).items() if v > 1]
+    path_to_node = {}
+    for node in module_attr_list:
+        path = node['path']
+        # Standard workaround: Catch inconsistency in Table C.36.8-1 where "Content Creator's Name" attribute
+        # appears twice in same hierarchy without a conditional
+        # http://dicom.nema.org/medical/dicom/2019c/output/chtml/part03/sect_C.36.8.html
+        if path not in DUPLICATE_PATH_EXCEPTIONS:
```
Comment (ReeceStevens) [inline dicom_standard/postprocess_merge_duplicate_nodes.py:37]: It seems like this might remove "Content Creator's Name" entirely, right? Wouldn't we want to keep at least one of the attributes?

File: dicom_standard/postprocess_merge_duplicate_nodes.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+def add_conditional_to_description(node):
+    conditional = node.get('conditional')
+    assert conditional is not None, f'Duplicate attribute (path: {node["path"]}) has no conditional statement.'
+    conditional = re.sub(r'\.$', ':', conditional)
+    formatted_conditional = f'<p style="font-weight: bold">{conditional[0].upper()}{conditional[1:]}</p>'
+    node['description'] = formatted_conditional + node['description']
+
+
+def merge_duplicate_nodes(module_attr_list):
```
Comment (ReeceStevens) [inline dicom_standard/postprocess_merge_duplicate_nodes.py:28]: I think this function is sufficiently complex that it would help to have some unit tests covering it, showing how the function works with simple mock data. That would also serve as a form of documentation, too.

File: dicom_standard/postprocess_merge_duplicate_nodes.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                    add_conditional_to_description(node)
+                    path_to_node[path]['description'] += node['description']
+                    path_to_node[path]['externalReferences'].extend(node['externalReferences'])
+            else:
+                if path in duplicate_paths:
+                    # Add conditional to description only if the duplicates do not have identical descriptions
+                    instances = filter(lambda n: n['path'] == path, module_attr_list)
+                    descriptions = map(lambda n: get_description_text(n['description']), instances)
+                    if len(set(descriptions)) > 1:
+                        add_conditional_to_description(node)
```
Comment (ReeceStevens) [inline dicom_standard/postprocess_merge_duplicate_nodes.py:52]: I feel like there is some duplication here that can be reduced. Is there a way this can be restructured to reduce the duplication of the filter/map and set comparison operations?

File: tests/end_to_end_test.py [hunk 4]
```diff
@@ -312,3 +312,10 @@ def test_no_duplicate_modules(self, modules):
 
     def test_no_duplicate_sops(self, sops):
         assert not self.get_duplicate_ids(sops)
+
+
+@pytest.mark.endtoend
+def test_no_duplicate_paths(module_attribute_relationship):
+    path_list = [d['path'] for d in module_attribute_relationship]
+    duplicate_paths = [k for k, v in Counter(path_list).items() if v > 1]
+    assert not duplicate_paths
```
Comment (ReeceStevens) [inline tests/end_to_end_test.py:321]: :+1: Good call adding this consistency check.

File: dicom_standard/postprocess_merge_duplicate_nodes.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    for node in module_attr_list:
+        path = node['path']
+        # Standard workaround: Catch inconsistency in Table C.36.8-1 where "Content Creator's Name" attribute
+        # appears twice in same hierarchy without a conditional
+        # http://dicom.nema.org/medical/dicom/2019c/output/chtml/part03/sect_C.36.8.html
+        if path not in DUPLICATE_PATH_EXCEPTIONS:
+            if path in path_to_node:
+                # Add conditional to description only if the duplicates do not have identical descriptions
+                instances = filter(lambda n: n['path'] == path, module_attr_list)
+                descriptions = map(lambda n: get_description_text(n['description']), instances)
```
Comment (ReeceStevens) [inline dicom_standard/postprocess_merge_duplicate_nodes.py:41]: It seems like we are only using `descriptions` to determine uniqueness. If that is true, do we need to wrap the description in `get_description_text` at all? Would the HTML strings still be identical in this case? I mention this only b/c HTML parsing is relatively slow and it may bog down this dedupe process if the HTML has to be parsed every time we do this uniqueness check.

It's possible there are some auto-generated ID fields that require you get the description this way, and that's totally fine-- just wanted to mention it in case there was an easier way.

Comment (russellkan) [reply dicom_standard/postprocess_merge_duplicate_nodes.py:41]: Good catch; I think this is obsolete. I initially did this because I had the merge logic in the preprocessing and the descriptions included some HTML `id` value that may have differed between identical descriptions. I think removing it entirely should be fine.

File: tests/postprocess_merge_duplicate_nodes_test.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
+import pytest
```
Comment (ReeceStevens) [inline tests/postprocess_merge_duplicate_nodes_test.py:1]: Great tests! These are clear, explicit, and make it obvious what the purpose of the function is.
